test/recipes/02-test_windows_registry.t

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/env perl
+# Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use OpenSSL::Test;
+use OpenSSL::Test::Utils;
+use OpenSSL::Test qw/:DEFAULT bldtop_file data_file srctop_file cmdstr/;
+
+setup("test_windows_registry");
+
+plan skip_all => "Windows registry tests are only available on windows"
+    if $^O !~ /(MSWin)/;
+
+my $actual;
+my $expect;
+
+my @tempout = run(app(["openssl", "version", "-w"]), capture => 1);
+my $context = "@tempout";
+$context =~ s/^.*: //;
+
+
+@tempout = run(app(["openssl", "version", "-v"]), capture => 1);
+my $version = "@tempout";
+$version =~ s/^OpenSSL //;
+$version =~ s/\.[0-9]\+-.*\n//;
```

Comment:
I'm quite confused by this regex. I can't quite see what the `\+` is doing. But AFAICT you are attempting to strip anything after the final patch number. But this seems to expect there to be an "-" as in "-dev"....but that doesn't appear in a production build. So this looks like it might start failing to do the right thing as soon as 3.4.0 is created (as opposed to "3.4.0-dev"). Or maybe I'm just confused.

Reply:
the regex is "find the pattern '.' followed by at least one of the digits 0-9 appearing at least once followed by a '-' and any number of other characters and a newline.

e.g 3.4.0-dev becomes 3.4, though I see what you're saying about branching stable releases